### PR TITLE
Cleaned up changelog and added resource spec link.

### PR DESCRIPTION
**Pull request by Yun-Ting**
## Changes
Combined these 2 PRs in the CHANGELOG.md.
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2949/
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2897
Unified changeLog format.
And added resource spec link to README.md.

### CODE REVIEW

File: docs/trace/extending-the-sdk/README.md [hunk 1]
```diff
@@ -317,3 +317,5 @@ A demo ResourceDetector is shown [here](./MyResourceDetector.cs).
   specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor)
 * [Sampler
   specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#sampler)
+* [Resource
```
Comment (reyang) [inline docs/trace/extending-the-sdk/README.md:320]: Minor: sort alphabetically (R before S) unless there is a good reason not to do so.

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,8 @@
 
 ## Unreleased
 
-* Publicize required methods to allow for custom implementations of resource
-  detectors.
+* Added support for implementing custom `ResourceDetector`.
   ([2949](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2949/))
```
Comment (reyang) [inline src/OpenTelemetry/CHANGELOG.md:6]: The links format need to be adjusted for consistency, search "1875" in this document and see how it was done.

**Review by reyang — APPROVED** [review]
LGTM assuming the comments will be addressed.

**Review by reyang — APPROVED** [review]
LGTM.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**